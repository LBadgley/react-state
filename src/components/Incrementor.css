p {
  background-color: skyblue;
  border: 4px dashed lightblue;
  width: 300px;
  text-align: center;
  font-size: 3em;
  color: white;
}

button {
  border: groove;
  border-radius: 15%;
  margin-left: 120px;
  font-size: 1em;
}
