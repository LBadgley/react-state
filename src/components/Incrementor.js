import React, { PureComponent } from 'react';
import styles from './Incrementor.css';


export default class Incrementor extends PureComponent {
  state = {
    count: 0
  };

  incrementCount = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  }
  render() {
    const { count } = this.state;
    return (
        <>
        <p style={styles}>Current count: {count}</p>
        <button style={styles} onClick={this.incrementCount}>Increment</button>
        </>
    );
  }
}

