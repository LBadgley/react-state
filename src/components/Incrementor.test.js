import React from 'react';
import { shallow } from 'enzyme';
import Incrementor from './Incrementor.js';

describe('incrementor test', () => {
  it('renders an incrementor', () => {
    const wrapper = shallow(<Incrementor />);
    expect(wrapper).toMatchSnapshot();
  }); 
});
