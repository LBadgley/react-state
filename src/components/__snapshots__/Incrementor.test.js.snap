// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incrementor test renders an incrementor 1`] = `
<Fragment>
  <p
    style={null}
  >
    Current count: 
    0
  </p>
  <button
    onClick={[Function]}
    style={null}
  >
    Increment
  </button>
</Fragment>
`;
